vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/MultiLayerWithPyramidDemo.java

```diff
@@ -160,34 +165,142 @@ private void createPolygonAndAddToActiveLayerGroup(final Building building)
 			.addTo(this.activeLayerGroup);
 	}
 	
-	record BuildingComplex(
-		String name,
-		List<Building> buildings
-	) implements BuildingData
+	final static class BuildingComplex implements BuildingData
 	{
+		private final String name;
+		private final List<Building> buildings;
+		
+		BuildingComplex(
+			String name,
+			List<Building> buildings
+		)
+		{
+			this.name = name;
+			this.buildings = buildings;
+		}
+		
+		public String name()
+		{
+			return name;
+		}
+		
+		public List<Building> buildings()
+		{
+			return buildings;
+		}
+		
+		@Override
+		public boolean equals(Object obj)
+		{
+			if(obj == this)
+			{
+				return true;
+			}
+			if(obj == null || obj.getClass() != this.getClass())
+			{
+				return false;
+			}
+			var that = (BuildingComplex)obj;
+			return Objects.equals(this.name, that.name) &&
+				   Objects.equals(this.buildings, that.buildings);
+		}
+		
+		@Override
+		public int hashCode()
+		{
+			return Objects.hash(name, buildings);
+		}
+		
+		@Override
+		public String toString()
+		{
+			return "BuildingComplex[" +
+				   "name=" + name + ", " +
+				   "buildings=" + buildings + ']';
+		}
+	
 		@Override
 		public List<BuildingData> nested()
 		{
 			return this.buildings()
 				.stream()
 				.map(BuildingData.class::cast)
-				.toList();
+				.collect(Collectors.toList());
 		}
 	}
 	
-	
-	record Building(
-		String name,
-		String url,
-		String imageUrl,
-		List<DataPoint> dataPoints
-	) implements BuildingData
+	final static class Building implements BuildingData
 	{
+		private final String name;
+		private final String url;
+		private final String imageUrl;
+		private final List<DataPoint> dataPoints;
+		
+		Building(
+			String name,
+			String url,
+			String imageUrl,
+			List<DataPoint> dataPoints
+		)
+		{
+			this.name = name;
+			this.url = url;
+			this.imageUrl = imageUrl;
+			this.dataPoints = dataPoints;
+		}
+		
```

Comment:
Many blank lines. But if the project rules do not forbid this, ok.

Reply:
Removed